src/data/step.puml

```diff
@@ -0,0 +1,36 @@
+@startuml
+skinparam activity {
+BackgroundColor #76CDD8
+ArrowColor black
+ArrowFontSize 13
+ArrowFontStyle bold
+ArrowFontName Segoe UI
+BorderColor black
+FontName Segoe UI
+FontSize 15
+}
+(*) --> "handle Step"
+-right-> "onStep"
+-left-> "handle Step"
+If "argument?"
+--> [Yes] "Process argument"
```

Comment:
After processing the step, there is no "process argument" step, it immediately checks the argument type